### ISSUE TITLE
readMSData error

### ISSUE THREAD

**Issue by jsaintvanne** [workflow4metabolomics/tools-metabolomics#164]
Hi,

Just running a mzML file from Metabolights with readMSData function and obtain the following error : 
```
Fatal error: Exit code 1 ()
Error in files[.hasSpecs] : invalid subscript type 'list'
Calls: readMSData
Execution halted
```

I tried on R and the file looks good and pass the function correctly : 
```
> readMSData("./dataTestMspurity/subset/POS/Urine_22_Top10_posneg_POS",mode="onDisk")
MSn experiment data ("OnDiskMSnExp")
Object size in memory: 1.43 Mb
- - - Spectra data - - -
 MS level(s): 1 2 
 Number of spectra: 3610 
 MSn retention times: 0:0 - 24:0 minutes
- - - Processing information - - -
Data loaded [Mon Dec  2 10:19:01 2019] 
 MSnbase version: 2.12.0 
- - - Meta data  - - -
phenoData
  rowNames: Urine_22_Top10_posneg_POS
  varLabels: sampleNames
  varMetadata: labelDescription
Loaded from:
  Urine_22_Top10_posneg_POS 
protocolData: none
featureData
  featureNames: F1.S0001 F1.S0002 ... F1.S3610 (3610 total)
  fvarLabels: fileIdx spIdx ... spectrum (35 total)
  fvarMetadata: labelDescription
experimentData: use 'experimentData(object)'
```

**Comment by jsaintvanne** [comment]
I found what was the problem... : 
- my files were named `ech1_POS` and were in mzML format
- I have got the previous error
- Because I have to name my files as following `ech1_POS.mzML`
- readMSData on Galaxy needs **ABSOLUTELY** the format in the name of the file !

It can be marked as "resolved" and be close !

**Comment by lecorguille** [comment]
Maybe XCMS change its behaviour about that but it was a time when XCMS detected the infiles according to a regular expression base on file extension.
Can you check that?
If indeed, both XCMS and readMSData don't need the extension anymore, we can adapt our wrappers.

**Comment by sneumann** [comment]
Hi, I was just bitten by this as well. I can confirm that 
`raw_data <- readMSData(files=c("L"), mode="onDisk")` works, where `L` is a mzML file. 
where exactly is that `files[.hasSpecs]` error coming from ? Ns that in the MSnbase package 
or in the W4M scripts in https://github.com/workflow4metabolomics/tools-metabolomics/blob/master/tools/msnbase_readmsdata/msnbase_readmsdata.r or in the `lib.r` scripts ? Yours, Steffen

**Comment by lecorguille** [comment]
Are we agree that the issue occur when you import file (.mzML) without extension, isn't it?

**Comment by sneumann** [comment]
Hi, MSnbase itself has no issues with `readMSData(files=c("L")` without extension as shown above.
Most important now is to debug where the `Error in files[.hasSpecs]` reported by the original poster is coming from. Is it because an issue on the xcms/MSnbase side ? Then I need a reproducible example. 
Or is it because of how W4M is making the call to `readMSData()` ? Ideally, I'd like to see a `traceBack()` of the error above. Yours, Steffen

**Comment by lecorguille** [comment]
@jsaintvanne Can you share your history with me?
I will be able to run the `traceBack()`

**Comment by lecorguille** [comment]
The issue come from this: https://github.com/workflow4metabolomics/tools-metabolomics/blob/master/tools/xcms_macro/lib.r#L350-L359

I wrote this code because historically, the xcms Galaxy wrappers took some zip. So I had to list the files that will be compatible with xcms. I got this code from the xmcs package itself. 

Since readMSData allows file without extension, I should adapt that by just detecting the file when we feed the tool with a zip file.
